src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -305,6 +403,86 @@ protected Object decode(
+                                    }
+                                    break;
+                                case 206:
+                                    position.set("diagnostic", buf.readUnsignedIntLE());
+                                    break;
+                                case 207:
+                                case 208:
+                                case 209:
+                                case 210:
+                                case 211:
+                                case 212:
+                                case 213:
+                                case 214:
+                                case 215:
+                                case 216:
+                                case 217:
+                                case 218:
+                                case 219:
+                                case 220:
+                                case 221:
+                                case 222:
+                                    position.set("user1Byte" + (j + 2 - 207), buf.readUnsignedByte());
+                                    break;
+                                case 223:
+                                case 224:
+                                case 225:
+                                case 226:
+                                case 227:
+                                case 228:
+                                case 229:
+                                case 230:
+                                case 231:
+                                case 232:
+                                case 233:
+                                case 234:
+                                case 235:
+                                case 236:
+                                case 237:
+                                    position.set("user2Byte" + (j + 2 - 223), buf.readUnsignedShortLE());
+                                    break;
+                                case 238:
+                                case 239:
+                                case 240:
+                                case 241:
+                                case 242:
+                                case 243:
+                                case 244:
+                                case 245:
+                                case 246:
+                                case 247:
+                                case 248:
+                                case 249:
+                                case 250:
+                                case 251:
+                                case 252:
+                                    position.set("user4Byte" + (j + 2 - 238), buf.readUnsignedIntLE());
+                                    break;
+                                case 253:
+                                case 254:
+                                case 255:
+                                    position.set("user8Byte" + (j + 2 - 253), buf.readLongLE());
+                                    break;
```

Comment:
These should probably be handled as generic IO values, similar to what we do with Teltonika for example.

Reply:
Are you talking about this design?
`position.set(Position.PREFIX_IO + id, readValue(buf, length));`

If yes, then it is more convenient for users who work with our configurator, documentation and who come to you from Wialon to look not at the field number, but at the size and ordinal number of the user parameter.

It's just that the field number is a lesser known value to the user.

Our version is more similar to the design for Galileo:
`position.set("can32BitR" + (tag - 0xdb), buf.readUnsignedIntLE());`
They also have these descriptions in the configurator and documentation.

I would suggest using the current option, what do you think?

Reply:
That should be fine, but I would still move these to default case, so we don't have these massive columns of cases.